src/components/calculator.js

```diff
@@ -1,13 +1,35 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
 import './calculator.css';
 import CalculatorKey from './calculatorkeys';
+import calculate from '../logic/calculate';
 
-const Calculator = () => (
-  <div className="calculator">
-    <DisplayResult />
-    <CalculatorKey />
+const Calculator = () => {
+  const [obj, setObj] = useState({ total: null, next: null, operation: null });
+  const handleClick = (buttonName) => {
+    const newObj = calculate(obj, buttonName);
+    setObj(newObj);
+  };
+  return (
+    <div className="calculator">
+      <DisplayResult result={obj.total} next={obj.next} operation={obj.operation} />
+      <CalculatorKey handleClick={handleClick} />
+    </div>
+  );
+};
+
+const DisplayResult = ({ result, next, operation }) => (
+  <div className="calculator__display">
+    <div>{next}</div>
+    {' '}
+    <div>{operation}</div>
+    <div>{result}</div>
   </div>
 );
 
-const DisplayResult = () => <div className="calculator__display">0</div>;
-
+DisplayResult.propTypes = {
+  result: PropTypes.string.isRequired,
+  next: PropTypes.string.isRequired,
+  operation: PropTypes.string.isRequired,
+};
```

Comment:
- [ ] good job @touredri, you can still make this function a separate component and import it here too as you did with `CalculatorKey` :+1:

Reply:
Yeah @VICTOR-CHUKWU good idea

Reply:
As it is not too long, cause I let there, interesting